Library/Homebrew/formula_support.rb

```diff
@@ -28,13 +28,14 @@ def valid?
 
   def to_s
     return @explanation unless @explanation.empty?
+    os_name = OS.mac? ? MacOS.os_name : 'macOS'
```

Comment:
That's because you need to not use `MacOS` (or `OS.mac?`) outside of `Library/Homebrew/os` or  `Library/Homebrew/extend/os/`. Instead, have an `OS.name` method which uses subclasses in `Library/Homebrew/extend/os/mac` to set the right macOS name.

Alternatively, edit `brew.sh` to set a `HOMEBREW_OS_NAME` (like `HOMEBREW_OS_VERSION` currently) which uses the correct version and then `OS.name` can just use `ENV["HOMEBREW_OS_NAME"]`

Shout if you need more help with this!